samcli/local/init/__init__.py

```diff
@@ -13,29 +13,53 @@
 
-RUNTIME_TEMPLATE_MAPPING = {
-    "python3.7": os.path.join(_templates, "cookiecutter-aws-sam-hello-python"),
-    "python3.6": os.path.join(_templates, "cookiecutter-aws-sam-hello-python"),
-    "python2.7": os.path.join(_templates, "cookiecutter-aws-sam-hello-python"),
-    "python": os.path.join(_templates, "cookiecutter-aws-sam-hello-python"),
-    "ruby2.5": os.path.join(_templates, "cookiecutter-aws-sam-hello-ruby"),
-    "nodejs6.10": os.path.join(_templates, "cookiecutter-aws-sam-hello-nodejs6"),
-    "nodejs8.10": os.path.join(_templates, "cookiecutter-aws-sam-hello-nodejs"),
-    "nodejs": os.path.join(_templates, "cookiecutter-aws-sam-hello-nodejs"),
-    "dotnetcore2.0": os.path.join(_templates, "cookiecutter-aws-sam-hello-dotnet"),
-    "dotnetcore2.1": os.path.join(_templates, "cookiecutter-aws-sam-hello-dotnet"),
-    "dotnetcore1.0": os.path.join(_templates, "cookiecutter-aws-sam-hello-dotnet"),
-    "dotnetcore": os.path.join(_templates, "cookiecutter-aws-sam-hello-dotnet"),
-    "dotnet": os.path.join(_templates, "cookiecutter-aws-sam-hello-dotnet"),
-    "go1.x": os.path.join(_templates, "cookiecutter-aws-sam-hello-golang"),
-    "go": os.path.join(_templates, "cookiecutter-aws-sam-hello-golang"),
-    "java8": os.path.join(_templates, "cookiecutter-aws-sam-hello-java"),
-    "java": os.path.join(_templates, "cookiecutter-aws-sam-hello-java")
+RUNTIME_DEP_TEMPLATE_MAPPING = {
+    "python3.7_pip": os.path.join(_templates, "cookiecutter-aws-sam-hello-python"),
+    "python3.6_pip": os.path.join(_templates, "cookiecutter-aws-sam-hello-python"),
+    "python2.7_pip": os.path.join(_templates, "cookiecutter-aws-sam-hello-python"),
+    "python_pip": os.path.join(_templates, "cookiecutter-aws-sam-hello-python"),
+    "ruby_bundler": os.path.join(_templates, "cookiecutter-aws-sam-hello-ruby"),
+    "ruby2.5_bundler": os.path.join(_templates, "cookiecutter-aws-sam-hello-ruby"),
+    "nodejs6.10_npm": os.path.join(_templates, "cookiecutter-aws-sam-hello-nodejs6"),
+    "nodejs8.10_npm": os.path.join(_templates, "cookiecutter-aws-sam-hello-nodejs"),
+    "nodejs_npm": os.path.join(_templates, "cookiecutter-aws-sam-hello-nodejs"),
+    "dotnetcore2.0_cli-package": os.path.join(_templates, "cookiecutter-aws-sam-hello-dotnet"),
+    "dotnetcore2.1_cli-package": os.path.join(_templates, "cookiecutter-aws-sam-hello-dotnet"),
+    "dotnetcore1.0_cli-package": os.path.join(_templates, "cookiecutter-aws-sam-hello-dotnet"),
+    "dotnetcore_cli_package": os.path.join(_templates, "cookiecutter-aws-sam-hello-dotnet"),
+    "dotnet_cli-package": os.path.join(_templates, "cookiecutter-aws-sam-hello-dotnet"),
+    "go1.x_go-get": os.path.join(_templates, "cookiecutter-aws-sam-hello-golang"),
+    "go_go-get": os.path.join(_templates, "cookiecutter-aws-sam-hello-golang"),
+    "java8_maven": os.path.join(_templates, "cookiecutter-aws-sam-hello-java-maven"),
+    "java_maven": os.path.join(_templates, "cookiecutter-aws-sam-hello-java-maven"),
```

Comment:
Do we need a `java_maven` and a `java8_maven`? This questions is a general question as we have shortcuts for every language which makes `sam init --help` super long and is only going to get worse if we have to shortcut all language + dependency manager combinations.

The code for maven and gradle look the same. Are they? If so, do we need to maintain two copies directly? Can we combine them so our java examples are the same but can have different dependency managers?

Reply:
Not really the same, slightly different source files, but the core java files are the same. We can look at de-duping that bit of source code.